SCons/Tool/linkCommon/SharedLibrary.py

```diff
@@ -135,10 +138,12 @@ def _get_shlib_stem(target, source, env, for_signature):
             % (target_name, shlibprefix, shlibsuffix)
         )
 
-    if target_name.startswith(shlibprefix):
-        target_name = target_name[len(shlibprefix) :]
+    if shlibprefix and target_name.startswith(shlibprefix):
```

Comment:
have a tiny testcase.. I'll add to comments.. To make this work for the symlinks, you need to move the shlibprefix after the shlibsuffix if.. logic.

But that doesn't address the initial target name which is still yielding `lib.so` (or on mac `lib.dylib`) instead of `liblib.so`